docker-compose.yaml

```diff
@@ -0,0 +1,41 @@
+version: '3'
+services: 
+  client:
+    image: node:12.18.4
+    ports:
+      - 8080:8080
+    volumes:
+      - type: bind
+        source: ./client
+        target: /code
+    working_dir: /code
+    command: bash -c "npm install && npm run dev"
+
+  api:
+    image: node:12.18.4
+    ports:
+      - 3000:3000
+    environment:
+      - DB_CONNECTION=mongodb://am-admin:am-pass@db:27017
+      - DB_NAME=lap2db
+    volumes:
+      - type: bind
+        source: ./server
+        target: /code
+    working_dir: /code
+    command: bash -c "npm install && npm run dev"
+    depends_on:
+      - db
+  
+  db:
+    image: mongo:latest
+    environment:
+      - MONGO_INITDB_ROOT_USERNAME=am-admin
+      - MONGO_INITDB_DATABASE=lap2db
+      - MONGO_INITDB_ROOT_PASSWORD=am-pass
+    volumes:
+      - "dbdata:/var/lib/mongodb/data" 
+      - "./db/seed.js:/docker-entrypoint-initdb.d/seed.js:ro"
+
+volumes:
+  dbdata:
```

Comment:
well done on setting up a YAML file for docker! I am assuming this is our common challenge?